### ISSUE TITLE
Does Formik not handle submit on enter?

### ISSUE THREAD

**Issue by IanBarbour** [formium/formik#1418]
Should I implement my own onKeyPress functionality or is there a way to handle this within the tools supplied in Formik?

Thanks!

**Comment by KidkArolis** [comment]
I was just solving a similar issue, I've included a visually hidden <input type="submit" tabindex="-1" /> element in my forms. This is when the default submit button is not within <form>, e.g. in a modal footer. Maybe that helps!

**Comment by KidkArolis** [comment]
* This way hitting enter anywhere in the form triggers submit event and Formik picks that up.

**Comment by jaredpalmer** [comment]
You maybe hitting the known input autofill React bug. Here is the umbrella issue: https://github.com/facebook/react/issues/11097

If you can provide a codesandbox we can then debug together

**Comment by jaredpalmer** [comment]
Generally tho, you should not implement onKeyPress

**Comment by justincy** [comment]
I ran into the same issue with a form in a dialog where the submit button was outside the form in the dialog footer. I fixed it the way @KidkArolis suggested with a hidden input.

**Comment by GomatoX** [comment]
I had a similar problem. Issue was that in my custom `Input` component I was passing props to the actual `<input {...props} />` element (it's not good). But actual reason was that props had formik form object so if you want to use spread you have to unsed *form* object from props. e.g.:

```js
const Input = ({ label, field, ...props }) => {
  delete props.form;

  return (
    <div className="some-class-name">
      <input {...field} {...props} />
    </div>
  );
};
```

or predefine all possible attributes manually

**Comment by kelly-tock** [comment]
I am not passing along the form prop into my custom input, I have a `type=submit` on the button, and am using the `<Form>` component. I tried manually doing onSubmit, onReset, also on the submit button, but hitting enter does not seem to fire any submit. clicking on the submit button does though, even when there is no click handler on it. 

😢

**Comment by kelly-tock** [comment]
with no custom input, my form is working with pressing enter. what else could cause this?  I have a custom on change as well, maybe something with the events?

**Comment by GomatoX** [comment]
> I am not passing along the form prop into my custom input, I have a `type=submit` on the button, and am using the `<Form>` component. I tried manually doing onSubmit, onReset, also on the submit button, but hitting enter does not seem to fire any submit. clicking on the submit button does though, even when there is no click handler on it.
> 
> 😢

What version of formik do you use?

**Comment by kelly-tock** [comment]
1.5.8

**Comment by kelly-tock** [comment]
I’ll post back when I figure it out. I started with the docs example in my form and am adding my customizations in one by one

**Comment by schmitch** [comment]
actually I have a textarea field, which does not work:

```
<Formik onSubmit={createComment} initialValues={formData} render={props =>
                <Form>
                    <div className="row my-2">
                        <div className="col-xxl-8 col-xl-6 col-12">
                            <h3>Kommentare</h3>
                        </div>
                        <div className="col-xxl-4 col-xl-6 col-12">
                            <button type="submit"
                                    className="btn btn-sm btn-success-flat float-right w-100"
                                    disabled={buttonDisabled}>
                                Hinzufügen
                            </button>

                        </div>
                    </div>

                    <div className="row mb-2">
                        <div className="col-12">
                            <Field component="textarea" rows="2" cols="20" className="form-control" name="value"
                                   required/>
                        </div>
                    </div>
                </Form>
```

**Comment by johnrom** [comment]
@schmitch I believe you have to do control+enter in order to get that working. Either way it's a bit different than this original issue which is whether the Formik component handles submit on Enter, while that refers to the specific field. Try asking your question in the Discord channel or opening a new issue.

**Comment by KSVarun** [comment]
> I was just solving a similar issue, I've included a visually hidden element in my forms. This is when the default submit button is not within
> 
> , e.g. in a modal footer. Maybe that helps!

Can you please let know how you did it. I am facing the same issue, where my submit is in my modal and pressing enter is not submitting the form

**Comment by steadylearner** [comment]
I had a problem with axios when used with Formik. It did continuously sent duplicate requests.

Then, I could find the problem while reading this issue. The problem came from the **sumbit="button" that I used to send request with Formik.

```jsx
<Button
   type="submit"
   disabled={isSubmitting}
   onClick={submitForm}
   fullWidth
   color="primary"
   variant="contained"
>
```

It is likely that when the type i set to "submit" it sends duplciate request.

So, I removed type="button" from the code and used the code similar to this.

```jsx
<Button
   type="submit"
   disabled={isSubmitting}
   onClick={submitForm}
   fullWidth
   color="primary"
   variant="contained"
>
<input className="x-display" type="submit" />
```

and CSS

```CSS
.x-display {
   display: none;
}
```

It solved both problem of duplicate request and using enter to submit the form.

First, I thought it was the proxy server problem that I was using but it was from the Formik.

Hope, this helps others later. I had to spend many times until I find this issue.

**Comment by timstallmann** [comment]
I'm running into this issue on a project as well and none of the solutions in the thread work -- is this still waiting on an upstream fix for https://github.com/facebook/react/issues/11097 ?

**Comment by johnrom** [comment]
@timstallmann submit on enter works. If that React bug is affecting you, you'll have to wait for that upstream, but generally enter triggers formik submit.

If you provide a codesandbox repro it'll be easier to discuss the issue you're having.

**Comment by bryndyment** [comment]
Make sure your input field is wrapped in `<Form> … </Form>` 😅

**Comment by zackdotcomputer** [comment]
I was running into this and got it fixed thanks to some tips in SO. The three things I found that affected this issue:

1. You need a `type="submit"` component. I recommend using an `input` because it worked most reliably for me.
2. Ideally, your submit component needs to be inside your `<form>` tags. I haven't tested the `form=` prop mentioned above.
3. *Important:* Your input needs to be visible when your `form` is painted by the browser. If you throw it in with 0-size or `display: none`, modern browsers won't turn on the "enter = submit" behavior we want. If you zero-size it with overflow: hidden *after* the component has mounted, though, you will still get the desired behavior as of Feb 2021...

**Comment by johnrom** [comment]
Yeah by hidden you'll definitely want to do something like [the screen reader text class](https://make.wordpress.org/accessibility/handbook/markup/the-css-class-screen-reader-text/) not only for reasons of Browser Compatibility, but so that screen readers will also understand there is a button that can be interacted with.

**Comment by ncoughlin** [comment]
> 
> 
> I am not passing along the form prop into my custom input, I have a `type=submit` on the button, and am using the `<Form>` component. I tried manually doing onSubmit, onReset, also on the submit button, but hitting enter does not seem to fire any submit. clicking on the submit button does though, even when there is no click handler on it.
> 
> 😢

You may be experiencing a similar issue to what I was, and I was able to solve it. My problem was that I had another button on the page that did not have a "type" defined. Therefore when pressing Enter the browser was finding the first button on the page, assumed it was the submit button because the type was not defined, and firing that instead. Make sure that all buttons on your page are defined as `type=button` except for your actual submit button.

**Comment by himanshuReact** [comment]
onClick={submitForm} this really helped
